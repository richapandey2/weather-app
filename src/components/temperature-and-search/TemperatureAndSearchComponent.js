import React from "react";
import { Box, Switch, TextField } from "@material-ui/core";
import PropTypes from "prop-types";
import { CELCIUS_ICON, FAHRENHEIT_ICON } from "../../constants/Constants";

function TempratureAndSearchComponent({
  isCelcius,
  handleSwitchButton,
  searchCity,
  handleTextOnChange,
  handleKeyPress,
}) {
  return (
    <Box className="div-box">
      <TextField
        id="standard-multiline-flexible"
        label="Search City"
        rowsMax={4}
        value={searchCity}
        onChange={handleTextOnChange}
        onKeyDown={handleKeyPress}
      />
      <div className="div-temp-box">
        {FAHRENHEIT_ICON}
        <Switch
          checked={isCelcius === true}
          onChange={(event) => handleSwitchButton(event)}
          color="default"
        />
        {CELCIUS_ICON}
      </div>
    </Box>
  );
}

TempratureAndSearchComponent.propTypes = {
  isCelcius: PropTypes.bool,
  handleSwitchButton: PropTypes.func,
  searchCity: PropTypes.string,
  handleTextOnChange: PropTypes.func,
  handleKeyPress: PropTypes.func,
};

export default TempratureAndSearchComponent;
