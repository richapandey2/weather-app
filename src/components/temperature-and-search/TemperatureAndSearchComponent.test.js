import React from "react";
import { render, cleanup } from "@testing-library/react";
import TemperatureAndSearchComponent from "./TemperatureAndSearchComponent";

afterEach(cleanup);

test("sholud test temperature and search component", () => {
  const { asFragment } = render(<TemperatureAndSearchComponent />);
  expect(asFragment).toMatchSnapshot();
});
