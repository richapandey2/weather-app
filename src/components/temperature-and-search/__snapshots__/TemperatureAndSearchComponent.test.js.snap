// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sholud test temperature and search component 1`] = `[Function]`;
